homeassistant/components/smart_meter_texas/get_ssl_context.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import re
+import socket
+import ssl
+import urllib
+
+import OpenSSL.crypto as crypto
+import certifi
+from smart_meter_texas.const import BASE_HOSTNAME
+
+_LOGGER = logging.getLogger(__name__)
+
+def get_ssl_context():
+    """Creates a usable SSL Context for Smart Meter Texas.
+    This will attempt to download and install the CA Issuers certificate into the SSL Context, then re-enable the SSL checking before proceeding.
+    """
+    ssl_context = None
+    try:
+        caiKey = "CA Issuers - URI:"
+        re_issuers_uri = re.compile(r"(https?://+[\w\d:#@%/;$()~_?\+-=\\\.&]*)", re.UNICODE)
+
+        ca_issuers_uri = None
+        ssl_context = ssl.create_default_context(capath=certifi.where())
+        ssl_context.check_hostname = False
+        ssl_context.verify_mode = ssl.CERT_NONE
+        with ssl_context.wrap_socket(socket.socket(), server_hostname=BASE_HOSTNAME) as s:
+            s.connect((BASE_HOSTNAME, 443))
+            cert_bin = s.getpeercert(True)
```

Comment:
This is doing I/O in the event loop.  It needs to run in the executor or use asyncio.open_connection

Reply:
I'll see what I can do to make this work with asyncio.

Reply:
https://docs.python.org/3/library/asyncio-protocol.html#asyncio.BaseTransport.get_extra_info

Looks like `get_extra_info` doesn't have an easy way to get the binary form so you might be stuck running it in the executor

Reply:
I think what I have in my current changes for the library should make the function call with the Socket connection run in the Executor... 

https://github.com/meyerrj/smart-meter-texas/blob/fc6d5fccafae59710b30db42d244778808c60796/smart_meter_texas/__init__.py#L361  

If you have a chance to review and let me know if you see anything that needs to be adjusted that would be helpful.

Reply:
Looks like that should do it 👍